src/app/route/CartPage/CartPage.container.js

```diff
@@ -118,12 +119,34 @@ export class CartPageContainer extends PureComponent {
             guest_checkout,
             showOverlay,
             showNotification,
-            device
+            device,
+            totals
         } = this.props;
 
         // to prevent outside-click handler trigger
         e.nativeEvent.stopImmediatePropagation();
 
+        const hasOutOfStockProductsInCart = totals.items.some((item) => {
```

Comment:
I believe this logic duplicates the one in CartOverlay. Consider separating this logic into an util and keeping it DRY.

Reply:
Okay